lib/logstash/filters/aggregate.rb

```diff
@@ -354,6 +438,32 @@ def filter(event)
 
   end
 
+  # Create a new event from the aggregation_map and the corresponding task_id
+  # This will create the event and
+  #  if @timeout_task_id_field is set, it will set the task_id on the timeout event
+  #  if @timeout_code is set, it will execute the timeout code on the created timeout event
+  # returns the newly created event
+  public 
+  def create_timeout_event(aggregation_map, task_id)
+    event_to_yield = LogStash::Event.new(aggregation_map)        
```

Comment:
Can you remove "public", because `create_timeout_event` method is private to plugin ?